buildSrc/src/main/java/com/scalar/db/JdkConfigurationPlugin.java

```diff
@@ -0,0 +1,142 @@
+package com.scalar.db;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+import javax.annotation.Nullable;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.tasks.compile.JavaCompile;
+import org.gradle.api.tasks.testing.Test;
+import org.gradle.jvm.toolchain.JavaLanguageVersion;
+import org.gradle.jvm.toolchain.JavaToolchainService;
+import org.gradle.jvm.toolchain.JvmVendorSpec;
+import org.jetbrains.annotations.NotNull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Plugin configuring compilation and integration test tasks to use a given SDK defined by the
+ * following Gradle properties:
+ *
+ * <ul>
+ *   <li>`compilationJavaVendor` : configure the SDK vendor used by compilation tasks
+ *   <li>`integrationTestJavaVersion` : configure all tasks name starting with "integrationTest" to
+ *       run with the given SDK version
+ *   <li>`integrationTestJavaVendor` : configure all tasks name starting with "integrationTest" to
+ *       run with the given SDK vendor
+ * </ul>
+ *
+ * Usage example using the CLI :
+ *
+ * <pre><code>
+ *   gradle integrationTestJdbc -PcompilationJavaVendor=amazon -PintegrationTestJavaVersion=11 -PintegrationJavaVendor=temurin
+ * </code></pre>
+ */
+public class JdkConfigurationPlugin implements Plugin<Project> {
+  private static final Logger logger = LoggerFactory.getLogger(JdkConfigurationPlugin.class);
+
+  private static final JavaLanguageVersion COMPILATION_JAVA_VERSION = JavaLanguageVersion.of(8);
```

Comment:
You are right; forcing the same Java version and vendor for compilation and building release makes sense. If so, we can remove `compilationJavaVendor`

Adding a `compilationJavaVersion` is a possibility, but in what case do you think it is useful? 
Do you want to provide this so that anybody can check out ScalarDB code without requiring Java 8 installation?

Reply:
> Adding a compilationJavaVersion is a possibility, but in what case do you think it is useful?

I have no idea for now. For example, we might want to build our CLI applications with GraalVM to shorten the bootstrap time in the future? I'm not sure, though. I think we can simply remove `compilationJavaVendor` for now.

Reply:
All right, thanks for your comment.
I will remove the `compilationJavaVendor` and require the compilation task to use JDK 8 (temurin).

Reply:
I revised it in https://github.com/scalar-labs/scalardb/pull/1749/commits/6869be5cc5673c960281f044475812ca45db1a88

Reply:
Sorry for the late reply.
Eventually, we will use JDK 11 for compilation, so enabling it to specify a version (and vendor) for compilation might be helpful. 
What do you think?

In that case, the following names might be more consistent, but this is a minor comment from my preference.
- `JavaCompilerVersion`
- `JavaCompilerVendor`
- `integrationTestJavaRuntimeVersion`
- `integrationTestJavaRuntimeVendor`

Reply:
You wish to set JDK 11 (temurin) as the default compiler, which can be overriden via the `javaCompilerVersion` and the `javaCompilerVendor` Gradle properties. Is my understanding correct?
I don't see a problem with making the compiler configurable, but as discussed above with Mitsu, we are not sure if there is really a use case for it now. Do you have something in mind?

Thank you for the naming suggestion. The naming is getting longer, but it's more consistent and correct, so I agree with it.

Reply:
@Torch3333 

> You wish to set JDK 11 (temurin) as the default compiler, which can be overriden via the javaCompilerVersion and the javaCompilerVendor Gradle properties. Is my understanding correct?

No. Using JDK 8 (temurin) by default is fine, but I would like the compilation version to be configurable so that we don't have to update the code and `ci.yaml` when we end the support for JDK 8, which might be coming soon (or might not be coming soon).
Of-course, we can update `JdkConfigurationPlugin.java` to do that, but we want to avoid it if it is simply avoidable.
What do you think?

Reply:
I see. In that case, to limit changes when we switch the compiler to Java 11, in my opinion, setting the compiler version in the `JdkConfigurationPlugin.java` without making it configurable with Gradle properties is the best course of action.
In any scenario, we can't avoid making changes, but not having a configurable compiler only requires slight changes when switching to Java 11.
 
If we compare the impact of making the JDK configurable or not and then switching to Java 11:

1. Without a configurable JDK, we need to:
- in the CI file, replace all mentions of JDK 8 by JDK 11
- update a single variable in the `JdkConfigurationPlugin.java`

2. With a configurable JDK, we need to :
- in the CI file, replace all mentions of JDK 8 by JDK 11
- we will need to add `-PjavaCompilerVersion=11 -PJavaCompilerVendor=temurin` every time we use gradle, which yields poor usability. We also need to add the same parameters for each gradle job in the CI too, which is a bit verbose.
- increase the `JdkConfigurationPlugin.java` complexity since we need to parse properties to configure the compiler.

Reply:
@feeblefakie @brfrn169 
Per our offline discussion, I will :
- make the compiler version and vendor configurable with Gradle properties
- refactor the CI to use global variables that define the compiler version and vendor to remove the hardcoded  usage of `Java 8 temurin`